packages/core/src/stream-loading/__tests__/stream-loader.test.ts

```diff
@@ -0,0 +1,129 @@
+import { jest } from '@jest/globals'
```

Comment:
this test file is not intended to be exhaustive.  I just happened to write a simple test at this layer while trying to figure out how to write a test at the lower StateManipulator layer.  Figured it out, but figured I'd leave this test in because why not.  I expect to more fully flesh out the tests for the StreamLoader in a subsequent PR though